src/ring_theory/witt_vector/structure_polynomial.lean

```diff
@@ -0,0 +1,137 @@
+```
+
+## Main results
+
+* `witt_structure_rat Φ`: the family of polynomials `ℕ → mv_polynomial (idx × ℕ) ℚ`
+  associated with `Φ : mv_polynomial idx ℚ` and satisfying the property explained above.
+* `witt_structure_rat_prop`: the proof that `witt_structure_rat` indeed satisfies the property.
+
+-/
+
+open mv_polynomial
+open set
+open finset (range)
+open finsupp (single)
+
+-- This lemma reduces a bundled morphism to a "mere" function,
+-- and consequently the simplifier cannot use a lot of powerful simp-lemmas.
+-- We disable this locally, and probably it should be disabled globally in mathlib.
+local attribute [-simp] coe_eval₂_hom
+
+variables {p : ℕ} {R : Type*} {idx : Type*} [comm_ring R]
+
+open_locale witt
+open_locale big_operators
+
+section p_prime
+
+variables (p) [hp : fact p.prime]
+include hp
+
+/-- `witt_structure_rat Φ` is a family of polynomials `ℕ → mv_polynomial (idx × ℕ) ℚ`
+that are uniquely characterised by the property that
+`bind₁ (witt_structure_rat p Φ) (witt_polynomial p ℚ n) = bind₁ (λ i, (rename (prod.mk i) (witt_polynomial p ℚ n))) Φ`.
+In other words: evaluating the `n`-th Witt polynomial on the family `witt_structure_rat Φ`
+is the same as evaluating `Φ` on the (appropriately renamed) `n`-th Witt polynomials.
+
+See `witt_structure_rat_prop` for this property,
+and `witt_structure_rat_exists_unique` for the fact that `witt_structure_rat`
+gives the unique family of polynomials with this property.
+
+These polynomials turn out to have integral coefficients,
+but it requires some effort to show this.
+See `witt_structure_int` for the version with integral coefficients,
+and `map_witt_structure_int` for the fact that it is equal to `witt_structure_rat`
+when mapped to polynomials over the rationals. -/
+noncomputable def witt_structure_rat (Φ : mv_polynomial idx ℚ) (n : ℕ) :
+  mv_polynomial (idx × ℕ) ℚ :=
+bind₁ (λ k, bind₁ (λ i, rename (prod.mk i) (W_ ℚ k)) Φ) (X_in_terms_of_W p ℚ n)
+
+theorem witt_structure_rat_prop (Φ : mv_polynomial idx ℚ) (n : ℕ) :
+  bind₁ (witt_structure_rat p Φ) (W_ ℚ n) =
+  bind₁ (λ i, (rename (prod.mk i) (W_ ℚ n))) Φ :=
+calc bind₁ (witt_structure_rat p Φ) (W_ ℚ n)
+    = bind₁ (λ k, bind₁ (λ i, (rename (prod.mk i)) (W_ ℚ k)) Φ) (bind₁ (X_in_terms_of_W p ℚ) (W_ ℚ n)) :
+      by { rw [bind₁_bind₁], apply eval₂_hom_congr (ring_hom.ext_rat _ _) rfl rfl }
+... = bind₁ (λ i, (rename (prod.mk i) (W_ ℚ n))) Φ :
+      by rw [bind₁_X_in_terms_of_W_witt_polynomial p _ n, bind₁_X_right]
+
+theorem witt_structure_rat_exists_unique (Φ : mv_polynomial idx ℚ) :
+  ∃! (φ : ℕ → mv_polynomial (idx × ℕ) ℚ),
+    ∀ (n : ℕ), bind₁ φ (W_ ℚ n) = bind₁ (λ i, (rename (prod.mk i) (W_ ℚ n))) Φ :=
+begin
+  refine ⟨witt_structure_rat p Φ, _, _⟩,
+  { intro n, apply witt_structure_rat_prop },
+  { intros φ H,
+    funext n,
+    rw show φ n = bind₁ φ (bind₁ (W_ ℚ) (X_in_terms_of_W p ℚ n)),
+    { rw [bind₁_witt_polynomial_X_in_terms_of_W p, bind₁_X_right] },
```

Comment:
Should this be a separate lemma? "If you can write `X` in terms of `W`, you can write family of polynomials `φ` in terms of `W`."

Reply:
I considered it. We had some lemmas like that, but they weren't used apart from this point. So I inlined them again.